storage/src/main/java/tech/pegasys/teku/storage/client/ChainHead.java

```diff
@@ -49,10 +49,14 @@ public UInt64 getForkChoiceEpoch() {
     return spec.computeEpochAtSlot(forkChoiceSlot);
   }
 
-  public UInt64 findCommonAncestor(final ChainHead other) {
+  public SlotAndBlockRoot findCommonAncestor(final ChainHead other) {
     if (getSlot().equals(UInt64.ZERO) || other.getSlot().equals(UInt64.ZERO)) {
       // One fork has no blocks so the only possible common ancestor is genesis.
-      return UInt64.ZERO;
+      if (getSlot().equals(UInt64.ZERO)) {
+        return new SlotAndBlockRoot(UInt64.ZERO, getRoot());
+      } else {
```

Comment:
I'd suggest removing the outer if case and making this
```suggestion
      } else if (other.getSlot().isZero()) {
```